### PR TITLE
Add basic vim bindings

### PR DESCRIPTION
**Pull request by Nimmidev**
I have added basic vim bindings for me a while ago. I was wondering if it would be something you are interested in having in the "main project" @Cloudef . I am happy to make any adjustments as needed (like removing the logs and adding keybinds to the README).

### CODE REVIEW

File: lib/vim.c [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <assert.h>
+
+#include "vim.h"
+
+
+#define LOG(x)
+//#define LOG(x) printf(x)
```
Comment (Cloudef) [inline lib/vim.c:12]: you may want to use fprintf(stderr, x) instead

File: client/common/common.c [hunk 2]
```diff
@@ -181,6 +181,7 @@ usage(FILE *out, const char *name)
           " -C, --no-cursor       ignore cursor events.\n"
           " -T, --no-touch        ignore touch events.\n"
           " -K, --no-keyboard     ignore keyboard events.\n"
+          " --vim                 use vim bindings.\n"
```
Comment (Cloudef) [inline client/common/common.c:184]: I think we should use --binding option with vim as argument instead. Some people may want to add their own bindings as well and there's already quite a lot of these options.

File: lib/vim.c (hunk 1 above)
Comment (Nimmidev) [reply lib/vim.c]: i removed the logging in 9aee56c3. I just used it for testing in the beginning.

File: client/common/common.c (hunk 2 above)
Comment (Nimmidev) [reply client/common/common.c]: Good call. I changed it in 6e1f83dc. I hope the implementation is satisfactory.

File: client/common/common.h [hunk 3]
```diff
@@ -36,7 +36,7 @@ struct client {
     bool force_fork, fork;
     bool no_exec;
     bool password;
-    bool use_vim_bindings;
+    char *key_binding;
```
Comment (Cloudef) [inline client/common/common.h:39]: Instead of char* can you make this enum?

**Comment by Cloudef** [comment]
Other than that it looks good. I think with this framework, even the original bindings could be refactored to own file. And perhaps even possibility for a implementation that implements customizable keybindings from a file.

File: client/common/common.h (hunk 3 above)
Comment (Nimmidev) [reply client/common/common.h]: Implemented in 51be5c55

**Comment by Nimmidev** [comment]
Yeah, that would be a good idea for the long term. Maybe i will look into this if i have more time available. We'll see.

**Comment by Cloudef** [comment]
Looks good thanks

**Comment by rieje** [comment]
Is this tested on Wayland? I'm on the latest version 0.6.14 with this feature and am not able to enter normal mode to test any of the differences offered with the setting. Another user [reported the same problem](https://github.com/Cloudef/bemenu/issues/315#issuecomment-1356244033) (I'm also on Arch and Wayland).

**Comment by Nimmidev** [comment]
@rieje see #315

**Comment by rieje** [comment]
Can [confirm](https://github.com/Cloudef/bemenu/issues/315#issuecomment-1485523651) it's working now.

Could you add <C-c> and <C-g> (and maybe even ESC) to terminate bemenu in normal mode? The first two already work in insert mode but don't in vim mode. It feels strange to have to go back to insert mode just to terminate bemenu.

**Pull request merged**